### PR TITLE
Add missing Famicon + Arcade (various) BIOS

### PR DESCRIPTION
**Pull request by lasers**
NES is missing Famicon BIOS in the database for Nestopia core.

> Nestopia requires the following BIOS image files for operation:
> 
>   NstDatabase.xml (NES Database)
>  DISKSYS.ROM (Famicom Disk System BIOS)

http://wiki.libretro.com/index.php?title=Nestopia

### CODE REVIEW

**Comment by hizzlekizzle** [comment]
Having disksys.rom in there makes sense, I think, but NstDatabase.xml only applies to Nestopia and not any of the other NES emulators. NstDatabase.xml also isn't strictly necessary for all games.

**Comment by lasers** [comment]
Not singling Nestopia out. I knew NES missed a bios for some time now... I think it may be not included intentionally because the idea is to extract everything to a single system directory. Popular choice too. First link I grabbed.

Just checked all NES cores...
NES requires disksys.rom (Famicon BIOS) for FCEUmm core too. 
Bnes and QuickNES does not support using BIOS image files.

And if you load Nestopia core, go to `Information` --> `Core Information`, you can see the status on the disksys.rom. It is `Missing, Required` for me because it's not in this database.

**Comment by RobLoach** [comment]
This is listed under Famicon Disk System `game` entry already:
https://github.com/lasers/libretro-database/blob/52665f0dffd978f296b03ca0a3b5363609d12bbd/dat/BIOS.dat#L83

```
game (
	name "Nintendo - Famicom Disk System"
	description "Nintendo - Famicom Disk System"
	rom ( name disksys.rom size 8192 crc 5e607dcf md5 ca30b50f880eb660a320674ed365ef7a sha1 57fe1bdee955bb48d357e463ccbf129496930b62 )
)
```

**Comment by lasers** [comment]
This is for https://github.com/libretro/RetroArch/issues/3725. As you suggested, I'm using the magic of `BIOS.dat` and clrmamepro to create a structure similar to what I proposed. This is what I have right now... with manual `disksys.rom` in NES directory. It's just not all there in the database.
```
~/.config/retroarch/bios
├── NEC - PC Engine - TurboGrafx 16
│   └── syscard3.pce
├── NEC - PC-FX
│   ├── pcfx.bios
│   └── pcfx.rom
├── Nintendo - Famicom Disk System
│   └── disksys.rom
├── Nintendo - Game Boy Advance
│   └── gba_bios.bin
├── Nintendo - Nintendo Entertainment System
│   ├── NstDatabase.xml
│   └── disksys.rom
├── Nintendo - Super Nintendo Entertainment System
│   ├── cx4.data.rom
│   ├── dsp1b.data.rom
│   ├── dsp1b.program.rom
│   ├── dsp2.data.rom
│   ├── dsp2.program.rom
│   ├── dsp3.data.rom
│   ├── dsp3.program.rom
│   ├── dsp4.data.rom
│   ├── dsp4.program.rom
│   ├── sgb.boot.rom
│   ├── st010.data.rom
│   ├── st010.program.rom
│   ├── st011.data.rom
│   ├── st011.program.rom
│   ├── st018.data.rom
│   └── st018.program.rom
├── Sega - Dreamcast
│   ├── dc_boot.bin
│   └── dc_flash.bin
├── Sega - Mega Drive - Genesis
│   ├── bios.gg
│   ├── bios_CD_E.bin
│   ├── bios_CD_J.bin
│   ├── bios_CD_U.bin
│   └── bios_J.sms
├── Sega - Saturn
│   ├── mpr-17933.bin
│   ├── saturn_bios.bin
│   └── sega_101.bin
└── Sony - PlayStation
    ├── scph5500.bin
    ├── scph5501.bin
    └── scph5502.bin

10 directories, 36 files
```
I'm not seeing anything in QuickNES/Bnes's .info files that need to be fixed. Also, is the libretro-super repo the place where all *.info comes from? In that case, I do want to fix Nestopia's corename.

**Comment by hizzlekizzle** [comment]
Yes, info files are all in the libretro-super repo, but that's not where a core's internal name comes from. That comes from the core's libretro.c/pp file. And, as radius(?) mentioned elsewhere, changing Nestopia to Nestopia UE would break a bunch of people's configs for very little gain. I agree that it would be more accurate but the cost/benefit just doesn't add up.

**Comment by lasers** [comment]
@hizzlekizzle I already made a pull request in nestopia-- but it'll be left up to libreto members to decide how to proceed from there. I agree it'll break... and I also mentioned the alternative...which is to fix info (long before I know how to find info... is which why I made the PR in first place).

**Comment by lasers** [comment]
Here's the PR https://github.com/libretro/nestopia/pull/34 -- I should have included that in previous post.

Radius was mentioning the breakage for per-cores when I suggested to use generic core config name for each core... like `~/.config/retroarch/config/NAME-OF-CORE/core.cfg` or `~/.config/retroarch/config/NAME-OF-CORE/config`

The one I'm trying to fix right now is to match up Nestopia / Nestopia UE one way or another.

**Comment by lasers** [comment]
Are we also missing the following bios?

Dinothawr requires loading dinothawr.game as a game ROM to work.
    dinothawr.game -- http://wiki.libretro.com/index.php?title=Dinothawr

~~Hatari requires a BIOS file in order to work (tos.img).
    tos.img -- http://wiki.libretro.com/index.php?title=Hatari~~

Final Burn Alpha lists a bio file (is it required or optional?)
    geo.rom -- http://wiki.libretro.com/index.php?title=Final_Burn_Alpha

**Comment by RobLoach** [comment]
> Are we also missing the following bios?

Looks like it! Mind sending PRs?

**Comment by lasers** [comment]
Sure if I find it. What about this? http://wiki.libretro.com/index.php?title=BlueMSX 

I'm little confused on that one as it does not say anything much.... and there are over 200 directories in `System/Machines`. Maybe we should wait until blueMSX libretro port get more feature supports first.

**Comment by lasers** [comment]
Dinothawr (core) won't run dinothawr.game without its assorted files.
I can't find geo.rom with that hash. It's either wrong or rare.

**Comment by RobLoach** [comment]
> Dinothawr (core) won't run dinothawr.game without its assorted files.

Is that considered a game bios? [dinothawr.game](https://github.com/libretro/Dinothawr/blob/master/dinothawr/dinothawr.game) looks like just an XML file. Does it live in RetroArch's bios folder?

> I can't find geo.rom with that hash. It's either wrong or rare.

It's possible the listed `neo-geo.rom` hash is incorrect. Do you have some other values available?

**Comment by lasers** [comment]
I guess the whole Dinothawr-data.zip could be more of a required asset files... similar to http://wiki.libretro.com/index.php?title=PPSSPP ... or we could category the whole thing as game bios. This also can be found under Online Updater --> Content Downloader --> Dinothawr so somebody could fix the wiki not to include any game bios/assets... and to tell them where it can be found in RetroArch.

It does not have to live in bios folder. I have it under `~/Games/Computer - Games/`. After loading the core (dinothawr), you need to load the content too (dinothawr.game). I'm just working with http://wiki.libretro.com/index.php?title=Dinothawr 

I'm on level 5-2 right now. Fight me irl.

File: dat/BIOS.dat [hunk 1]
```diff
@@ -93,6 +99,7 @@ game (
 	name "Nintendo - Nintendo Entertainment System"
 	description "Nintendo - Nintendo Entertainment System"
 	rom ( name NstDatabase.xml size 1009534 crc ebb2196c md5 7bfe8c0540ed4bd6a0f1e2a0f0118ced sha1 26322f182540211e9b5e3647675b7c593706ae2b )
+	rom ( name disksys.rom size 8192 crc 5e607dcf md5 ca30b50f880eb660a320674ed365ef7a sha1 57fe1bdee955bb48d357e463ccbf129496930b62 )
```
Comment (RobLoach) [inline dat/BIOS.dat]: disksys.rom is already apart of the Famicon entry.

Comment (lasers) [inline dat/BIOS.dat]: I understand disksys.rom comes from Famicon system.

There are two NES cores that uses Famicon right now.

http://wiki.libretro.com/index.php?title=Nintendo_NES_Core_Compatibility

>"Note: Nestopia and FCEUmm are the only cores with fully functional Famicom Disk System support."

The wikipedia page also have both of them under NES too.

https://en.wikipedia.org/wiki/Nintendo_Entertainment_System

>At June 1985's Consumer Electronics Show (CES), Nintendo unveiled the American version of its Famicom, with a new case redesigned by Lance Barr and featuring a "zero insertion force" cartridge slot.[22] This is the system which would eventually be officially deployed as the Nintendo Entertainment System, or the colloquial "NES".

They're pretty much same... I see that we do not have a Famicon core so if you're going to play some Famicon games, you will have to use one of two NES cores.

According to the display_name in all NES/Famicom cores (bnes, emux_nes, fceumm, nestopia, and quicknes), they contains `NES / Famicom (coreName)`.

According to database in `fceumm_libretro.info`, `nestopia_libretro.info`, and `quicknes_libretro.info`, maybe the name should be... `Nintendo - Nintendo Entertainment System|Nintendo - Famicom Disk System` although I'm not sure how clrmamepro and other rom managers would react to that. 😄 

How should we solve this?

**Pull request closed without merging**